custom_components/bosch_alarm/switch.py

```diff
@@ -48,9 +50,14 @@ async def async_turn_off(self, **kwargs):
 async def async_setup_entry(hass, config_entry, async_add_entities):
     """Set up switch entities for outputs"""
 
-    panel = hass.data[DOMAIN][config_entry.entry_id]
-    device_info = device_info_from_panel(panel)
-    async_add_entities(
-            PanelOutputEntity(id, output, device_info, panel)
+    data = hass.data[DOMAIN][config_entry.entry_id]
+    panel = data.panel
+
+    def setup():
+        device_info = device_info_from_panel(data)
```

Comment:
If we're passing in all of data into the constructor anyway, we might as well have the constructor get the device info itself